src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeParameterInfo.cs

```diff
@@ -337,46 +346,17 @@ public override bool HasDefaultValue
             }
             #endregion
 
+            // If default value is not specified in metadata, look for it in custom attributes
             if (defaultValue == DBNull.Value)
             {
-                #region Look for a default value in the custom attributes
-                if (raw)
-                {
-                    foreach (CustomAttributeData attr in CustomAttributeData.GetCustomAttributes(this))
-                    {
-                        Type? attrType = attr.Constructor.DeclaringType;
-
-                        if (attrType == typeof(DateTimeConstantAttribute))
-                        {
-                            defaultValue = GetRawDateTimeConstant(attr);
-                        }
-                        else if (attrType == typeof(DecimalConstantAttribute))
-                        {
-                            defaultValue = GetRawDecimalConstant(attr);
-                        }
-                        else if (attrType!.IsSubclassOf(typeof(CustomConstantAttribute)))
-                        {
-                            defaultValue = GetRawConstant(attr);
-                        }
-                    }
-                }
-                else
-                {
-                    object[] CustomAttrs = GetCustomAttributes(typeof(CustomConstantAttribute), false);
-                    if (CustomAttrs.Length != 0)
-                    {
-                        defaultValue = ((CustomConstantAttribute)CustomAttrs[0]).Value;
-                    }
-                    else
-                    {
-                        CustomAttrs = GetCustomAttributes(typeof(DecimalConstantAttribute), false);
-                        if (CustomAttrs.Length != 0)
-                        {
-                            defaultValue = ((DecimalConstantAttribute)CustomAttrs[0]).Value;
-                        }
-                    }
-                }
-                #endregion
+                // The resolution of default value is done by following these rules:
+                // 1. For RawDefaultValue, we pick the first custom attribute holding the constant value
+                //  in the following order: DecimalConstantAttribute, DateTimeConstantAttribute, CustomConstantAttribute
+                // 2. For DefaultValue, we first look for CustomConstantAttribute and pick the first occurrence.
+                //  If none is found, then we repeat the same process searching for DecimalConstantAttribute.
+                // IMPORTANT: Please note that there is a subtle difference in order custom attributes are inspected for
+                //  RawDefaultValue and DefaultValue.
```

Comment:
```suggestion
                // The resolution of default value is done by following these rules:
                // 1. For RawDefaultValue, we pick the first custom attribute holding the constant value
                //  in the following order: DecimalConstantAttribute, DateTimeConstantAttribute, CustomConstantAttribute
                // 2. For DefaultValue, we first look for CustomConstantAttribute and pick the first occurrence.
                //  If none is found, then we repeat the same process searching for DecimalConstantAttribute.
                // IMPORTANT: Please note that there is a subtle difference in order custom attributes are inspected for
                //  RawDefaultValue and DefaultValue.
```
These two comments would me more appropriate in the GetDefaultValueFromCustomAttributeData and GetDefaultValueFromCustomAttributes methods

Reply:
I placed the comments at the call site in all runtime variants, since the calling methods are not exactly the same between the runtimes.